DataDog/helm-charts
# Allow updateMap to use provided sourceFile and mappingFile

#### What this PR does / why we need it:

Add support to the `-updateMap` flag to update provided `mappingFile` with a provided `sourceFile` so that it can be used more generically in development and CI. 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
**QA**
* Uncomment line `295` to print tmp file paths  and build yaml-mapper binary 
* `./yaml-mapper -updateMap` with no flags should update default mappingFile with values from the latest Datadog chart values.yaml: 
  * Remove a few unmapped helm keys from `mapping_datadog_helm_to_datadogagent_crd.yaml` and run `./yaml-mapper -updateMap`. The helm keys should be added back to the default mapping file. (i.e. remove `agents.containers.agent.envDict: ""`)
  * temp chart.yaml and datadog-values.yaml files should get cleaned up 
* `./yaml-mapper -updateMap -sourceFile=<path-to-local-helm-charts-dir>/charts/datadog/values.yaml` should update the default mappingFile with any new keys added to the local values.yaml file
* `./yaml-mapper -updateMap -sourceFile=<path-to-local-helm-charts-dir>/charts/datadog/values.yaml -mappingFile=<other-mapping-file>` should update provided mappingFile with keys from the local values.yaml file
* `./yaml-mapper -updateMap -sourceFile=<any.yaml> -mappingFile=<some-mapping-file>` should map keys from any yaml file to a provided mapping file
* For the above testing where `-sourceFile` flag is used , make sure provided `sourceFile` doesn't get deleted!
* values.yaml => DDA mapping should be unaffected

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
